cmd/waitshutdown/main.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"time"
+
+	"k8s.io/ingress-nginx/internal/nginx"
+)
+
+func main() {
+	// wait for the NGINX process to terminate
+	timer := time.NewTicker(time.Second * 1)
+	for range timer.C {
+		if !nginx.IsRunning() {
+			timer.Stop()
+			break
+		}
+	}
```

Comment:
Should we also give time for controller to exit?

Reply:
> Should we also give time for controller to exit?

Not really. The controller is already stopped and just waiting for nginx https://github.com/kubernetes/ingress-nginx/blob/82b241c5175bfd226658527156d11a15e0e78039/internal/ingress/controller/nginx.go#L415
The only goal of this new binary is to delay the termination of the pod until there is no nginx process running.